source/core/index.ts

```diff
@@ -727,6 +727,11 @@ export default class Request extends Duplex implements RequestEvents<Request> {
 				const redirectBuffer = Buffer.from(response.headers.location, 'binary').toString();
 				const redirectUrl = new URL(redirectBuffer, url);
 
+				if ((url as URL).hostname !== 'unix' && (redirectUrl.protocol === 'unix:' || redirectUrl.hostname === 'unix')) {
```

Comment:
I think this would read better if we made a `isUnixDomainSocketURL` function and used it for both `url` and `redirectUrl`.

Reply:
PTAL